add ACI worker guide (#12846)
PrefectHQ/prefect: docs/integrations/prefect-azure/aci_worker.md

```bash
export RG_NAME=<resource-group-name> && \
az group create --name $RG_NAME --location <location>
```

Throughout the rest of the guide, we'll need to refer to the **scope** of the created resource group, which is a string describing where the resource group lives in the hierarchy of your Azure account. To save the scope of your resource group as an environment variable, run the following command:

```bash
RG_SCOPE=$(az group show --name $RG_NAME --query id --output tsv)
```

You can check that the scope is correct before moving on by running `echo $RG_SCOPE` in your terminal. It should be formatted as follows:

```
/subscriptions/<subscription-id>/resourceGroups/<resource-group-name>
```

## Step 2. Prepare ACI permissions

In order for the worker to create, monitor, and delete the other Container Instances in which flows will run, we'll need to create a **custom role** and an **identity**, and then affiliate that role to the identity with a **role assignment**. When we start our worker, we'll assign that identity to the Container Instance it's running in.

### 1. Create a role

> The custom `Container Instances Contributor` role has all the permissions your worker will need to run flows in other Container Instances. Create it by running the following command:

Reviewer: ```suggestion
The custom `Container Instances Contributor` role has all the permissions your worker will need to run flows in other container instances. Create it by running the following command:
```

Reviewer: I think this should be lower case - but maybe they always capitalize it, in which case, please disregard the suggested change.

  ]
}'
```

### 2. Create an identity

Create a user-managed identity with the following command, replacing `<identity-name>` with the name you'd like to use for the identity:

```bash
export $IDENTITY_NAME=<identity_name> && \
az identity create -g $RG_NAME -n $IDENTITY_NAME
```

We'll also need to save the principal id and full object id of the identity for the role assignment and container creation steps, respectively:

```bash
IDENTITY_PRINCIPAL_ID=$(az identity list --query "[?name=='$IDENTITY_NAME'].principalId" --output tsv) && \
IDENTITY_ID=$(az identity list --query "[?name=='$IDENTITY_NAME'].id" --output tsv)
```

### 3. Assign roles to the identity

Now let's assign the `Container Instances Contributor` role we created earlier to the new identity:

```bash
az role assignment create \
    --assignee $IDENTITY_ID \
    --role "Container Instances Contributor" \
    --scope $RG_SCOPE
```

Since we'll be using ACR to host a custom Docker image containing a Prefect flow later in the guide, let's also assign the built in `AcrPull` role to the identity:

```bash
az role assignment create \
    --assignee $IDENTITY_ID \
    --role "AcrPull" \
    --scope $RG_SCOPE
```

> ## Step 3. Create the worker Container Instance

Reviewer: sSame note as above on caps. I assume intentional.

## Step 4. Create an ACR registry

In order to build and push images containing flow code to Azure, we'll need a container registry. Create one with the following command, replacing `<container-registry-name>` with the registry name of your choosing:

```bash
az acr create --resource-group $RG_NAME \
  --name <container-registry-name> --sku Basic
```

## Step 5. Update your ACI work pool configuration

Once your work pool is created, navigate to the Edit page of your ACI work pool. You will need to update the following fields:

### Identities

This will be your `IDENTITY_ID`. You can get it from your terminal by running `echo $IDENTITY_ID`. When adding it to your work pool, it should be formatted as a JSON array:

```
["/subscriptions/<subscription-id>/resourcegroups/<resource-group-name>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<identity-name>"]
```

![Configuring an ACI work pool's identities.](/img/guides/aci-worker-identities.png)

### ACRManagedIdentity

> This is required for your flow code containers to be pulled from ACR. It consists of

Reviewer: ```suggestion
ACRManagedIdentity is required for your flow code containers to be pulled from ACR. It consists of the following:
```

- Identity: the same `IDENTITY_ID` as above, as a string
```
/subscriptions/<subscription-id>/resourcegroups/<resource-group-name>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<identity-name>
```
- Registry URL: your `<registry-name>`, followed by `.azurecr.io`
```
<registry-name>.azurecr.io
```

![Configuring an ACI work pool's ACR Managed Identity.](/img/guides/aci-worker-acridentity.png)

> ### Subscription ID and Resource Group Name

Reviewer: ```suggestion
### Subscription ID and resource group name
```

> Both the subscription ID and resource group name can be found in in the `RG_SCOPE` environment variable created earlier in the guide, viewed by running `echo $RG_SCOPE`:

Reviewer: ```suggestion
Both the subscription ID and resource group name can be found in in the `RG_SCOPE` environment variable created earlier in the guide. View their values by running `echo $RG_SCOPE`:
```

```
/subscriptions/<subscription-id>/resourceGroups/<resource-group-name>
```

![Configuring an ACI work pool.](/img/guides/aci-worker-subscription.png)

Then click Save.

## Step 6. Pick up a flow run with your new worker

> This guide uses ACR to store a Docker image containing your flow code. To do this, we will write a flow, then deploy it using `flow.deploy()`, which will copy flow code into a Docker image and push that image to an ACR registry.

Reviewer: ```suggestion
This guide uses ACR to store a Docker image containing your flow code. Write a flow, then deploy it using `flow.deploy()`, which will copy flow code into a Docker image and push that image to an ACR registry.
```
